modules/taboolaBidAdapter.js

```diff
@@ -0,0 +1,250 @@
+
+/**
+ * try to extract User Id by that order:
+ *  local storage
+ *  first party cookie
+ *  rendered trc
+ *  new user set it to 0
+ */
+export const userData = {
+  storageManager: getStorageManager(GVLID, BIDDER_CODE),
+  getUserId: () => {
+    const {getFromLocalStorage, getFromCookie, getFromTRC} = userData;
+
+    try {
+      return getFromLocalStorage() || getFromCookie() || getFromTRC();
+    } catch (ex) {
+      return 0;
+    }
+  },
+  getFromCookie() {
+    const {cookiesAreEnabled, getCookie} = userData.storageManager;
+    if (cookiesAreEnabled()) {
+      const cookieData = getCookie(COOKIE_KEY);
+      const userId = userData.getCookieDataByKey(cookieData, USER_ID);
+      if (userId) {
+        return userId;
+      }
+    }
+  },
+  getCookieDataByKey(cookieData, key) {
+    const [, value = ''] = cookieData.split(`${key}=`)
+    return value;
+  },
+  getFromLocalStorage() {
+    const {hasLocalStorage, localStorageIsEnabled, getDataFromLocalStorage} = userData.storageManager;
+
+    if (hasLocalStorage() && localStorageIsEnabled()) {
+      return getDataFromLocalStorage(STORAGE_KEY);
+    }
+  },
+  getFromTRC() {
+    return window.TRC ? window.TRC.user_id : 0;
+  }
+}
+
+export const internal = {
+  getPageUrl: (refererInfo = {}) => {
+    if (refererInfo.canonicalUrl) {
+      return refererInfo.canonicalUrl;
+    }
+
+    if (config.getConfig('pageUrl')) {
+      return config.getConfig('pageUrl');
+    }
+
+    try {
+      return getWindowTop().location.href;
+    } catch (e) {
+      return getWindowSelf().location.href;
+    }
+  },
+  getReferrer: (refererInfo = {}) => {
+    if (refererInfo.referer) {
+      return refererInfo.referer;
+    }
+
+    try {
+      return getWindowTop().document.referrer;
+    } catch (e) {
+      return getWindowSelf().document.referrer;
+    }
+  }
+}
+
+export const spec = {
+  supportedMediaTypes: [BANNER],
+  gvlid: GVLID,
+  code: BIDDER_CODE,
+  isBidRequestValid: (bidRequest) => {
+    return !!(bidRequest.sizes &&
+           bidRequest.params &&
+           bidRequest.params.publisherId &&
+           bidRequest.params.tagId);
+  },
+  buildRequests: (validBidRequests, bidderRequest) => {
+    const [bidRequest] = validBidRequests;
+    const {refererInfo, gdprConsent = {}, uspConsent} = bidderRequest;
+    const {bcat = [], badv = [], publisherId} = bidRequest.params;
+    const site = getSiteProperties(bidRequest.params, refererInfo);
+    const device = {ua: navigator.userAgent};
+    const imps = getImps(validBidRequests);
+    const user = {
+      buyeruid: userData.getUserId(gdprConsent, uspConsent),
+      ext: {}
+    };
+    const regs = {
+      coppa: 0,
+      ext: {}
+    };
+
+    if (gdprConsent.gdprApplies) {
+      user.ext.consent = bidderRequest.gdprConsent.consentString;
+      regs.ext.gdpr = 1;
+    }
+
+    if (uspConsent) {
+      regs.ext.us_privacy = uspConsent;
+    }
+
+    if (config.getConfig('coppa')) {
```

Comment:
don't you want to use this value?

Reply:
it's boolean if it's set than it's means  `coppa` enabled so no need